packages/compiler/core/diagnostics.ts

```diff
@@ -164,7 +175,12 @@ export function getSourceLocation(
     return createDummySourceLocation(target.path);
   }
 
+  if (target.kind === "Function" || target.kind === "function" || target.kind === "Object") {
```

Comment:
I think this due to mixing different discriminated types. it can either be the symbol "function" or the node kind: "Function" but I agree this is a bit confusing

Reply:
Ah, probably best to use same conventional casing for discriminators and give them unique names.